### PR TITLE
Deprecate metrics from opentelemetry-api, add a global version in metrics api

### PR DESCRIPTION
**Pull request by bogdandrutu**
One small caveat, the GlobalMeterProvider class may be removed when we do the stable release. When the metrics will be stable/release candidate we need to find a different solution for that class in order to avoid removing it just before the release.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: exporters/otlp/src/main/java/io/opentelemetry/exporter/otlp/OtlpGrpcSpanExporter.java [hunk 1]
```diff
@@ -105,6 +99,12 @@
    *     or to a negative value, the exporter will wait indefinitely.
    */
   private OtlpGrpcSpanExporter(ManagedChannel channel, long deadlineMs) {
+    Meter meter = GlobalMetricsProvider.getMeter("io.opentelemetry.exporters.otlp");
+    this.spansSeen =
+        meter.longCounterBuilder("spansSeenByExporter").build().bind(EXPORTER_NAME_LABELS);
+    LongCounter spansExportedCounter = meter.longCounterBuilder("spansExportedByExporter").build();
+    this.spansExportedSuccess = spansExportedCounter.bind(EXPORT_SUCCESS_LABELS);
+    this.spansExportedFailure = spansExportedCounter.bind(EXPORT_FAILURE_LABELS);
```
Comment (jkwatson) [inline exporters/otlp/src/main/java/io/opentelemetry/exporter/otlp/OtlpGrpcSpanExporter.java:107]: I thought we had already done this. Good catch!

**Review by jkwatson — APPROVED** [review]
Thanks!

File: exporters/otlp/src/main/java/io/opentelemetry/exporter/otlp/OtlpGrpcSpanExporter.java (hunk 1 above)
Comment (bogdandrutu) [reply exporters/otlp/src/main/java/io/opentelemetry/exporter/otlp/OtlpGrpcSpanExporter.java:107]: I need to unbind them in shutdown.

**Comment by jkwatson** [comment]
looks like this needs a rebase, @bogdandrutu

File: exporters/otlp/src/main/java/io/opentelemetry/exporter/otlp/OtlpGrpcSpanExporter.java (hunk 1 above)
Comment (bogdandrutu) [reply exporters/otlp/src/main/java/io/opentelemetry/exporter/otlp/OtlpGrpcSpanExporter.java:97]: Done.

**Comment by bogdandrutu** [comment]
@anuraaga does this look good for you?

**Pull request merged**